src/frequenz/sdk/actor/_run_utils.py

```diff
@@ -38,18 +38,22 @@ async def run(*actors: Actor) -> None:
         done_tasks, pending_tasks = await asyncio.wait(
             pending_tasks, return_when=asyncio.FIRST_COMPLETED
         )
-
         # This should always be only one task, but we handle many for extra safety
         for task in done_tasks:
-            # Cancellation needs to be checked first, otherwise the other methods
-            # could raise a CancelledError
-            if task.cancelled():
+            # BackgroundService returns a BaseExceptionGroup containing multiple
+            # exceptions. The 'await' statement raises these exceptions, and 'except*'
+            # is used to handle them as a group. If the task raises multiple different
+            # exceptions, 'except*' will be invoked multiple times, once for each
+            # exception group.
+            try:
+                await task
+            except* asyncio.CancelledError:
                 _logger.info("Actor %s: Cancelled while running.", task.get_name())
-            elif exception := task.exception():
+            except* BaseException as err:
                 _logger.error(
                     "Actor %s: Raised an exception while running.",
                     task.get_name(),
-                    exc_info=exception,
+                    exc_info=err,
```

Comment:
You could change this with `_logger.exception()` now, as it is inside a `try/except`.

Reply:
done